types/markdownlint/index.d.ts

```diff
@@ -18,7 +18,233 @@ declare namespace markdownlint {
     function readConfigSync(file: string): MarkdownlintConfig;
 
     interface MarkdownlintConfig {
-        [rule: string]: boolean | { [key: string]: any };
+        [rule: string]: undefined | string | number | boolean | { [key: string]: any };
+
+        /**
+         * Default state for all rules.
+         * @default true
+         */
+        default?: boolean;
+
+        /**
+         * Heading levels should only increment by one level at a time.
+         *
+         * This rule is triggered when you skip heading levels in a markdown document.
+         */
+        "heading-increment"?: boolean;
+        /**
+         * Heading levels should only increment by one level at a time.
+         *
+         * This rule is triggered when you skip heading levels in a markdown document.
+         */
+        "header-increment"?: boolean;
+
+        /**
+         * First heading should be a top level heading.
+         * @deprecated
+         * @see {"first-line-heading"}
+         * @see {"first-line-h1"}
+         */
+        "first-heading-h1"?: boolean | {
+            level?: number;
+        };
+        /**
+         * First heading should be a top level heading.
+         * @deprecated
+         * @see {"first-line-heading"}
+         * @see {"first-line-h1"}
+         */
+        "first-header-h1"?: MarkdownlintConfig["first-heading-h1"];
+
+        /**
+         * Heading style.
+         * @default "consistent"
+         */
+        "heading-style"?: {
+            style?: "consistent" | "atx" | "atx_closed" | "setext" | "setext_with_atx" | "setext_with_atx_closed"
+        };
+        /**
+         * Heading style.
+         * @default "consistent"
+         */
+        "header-style"?: MarkdownlintConfig["heading-style"];
+
+        /**
+         * This rule is triggered when the symbols used in the document for unordered list items do not match the configured unordered list style:
+         *
+         * ```md
+         * * Item 1
+         * + Item 2
+         * - Item 3
+         * ```
+         *
+         * @default "consistent"
+         */
+        "ul-style"?: {
+            style?: "consistent" | "asterisk" | "plus" | "dash" | "sublist";
+        };
+
+        /**
+         * Inconsistent indentation for list items at the same level
+         */
+        "list-indent"?: boolean;
+
+        /**
+         * Consider starting bulleted lists at the beginning of the line.
+         */
+        "ul-start-left"?: boolean;
+
+        /**
+         * This rule is triggered when list items are not indented by the configured number of spaces.
+         * @default 2
+         */
+        "ul-indent"?: boolean | {
+            indent?: number;
+        };
+
+        "trailing-spaces"?: boolean | {
```

Comment:
I think it should be
```suggestion
        "no-trailing-spaces"?: boolean | {
```